openid-4-verifiable-presentations-1_0.md

```diff
@@ -283,6 +296,17 @@ The following is a non-normative example of an Authorization Request:
     &nonce=n-0S6_WzA2Mj HTTP/1.1
 ```
 
+The following is a non-normative example with a `request_uri_method` parameter (including the additional parameters `client_id_scheme` and `client_metadata`): 
+
+```
+  GET /authorize?
+    client_id=client.example.org
+    &client_id_scheme=x509_san_dns
+    &client_metadata=...
+    &request_uri=https%3A%2F%2Fclient.example.org%2Frequest
```

Comment:
request_uri needs to be unique per request I think:

```suggestion
    &request_uri=https%3A%2F%2Fclient.example.org%2Frequest%2Fvapof4ql2i7m41m68uep
```

Reply:
WDYT about this:

### attack scenario
An adversary that has obtained/steal the RP's private key would be able to issue its own signed requests at its own request_uri endpoint, as well as hijack presentations to some evil response_uri endpoint.

### solution

to prevent hijack attacks it would be better that the RP exposes the request_uri in its metadata and this metadata should be signed by a trusted third party. Having obtained the signed metadata, a wallet instance will use the attested endpoints.

### issue
See also: https://github.com/openid/OpenID4VP/issues/73 where I was suggesting using fragments to allow static uris in the metadata with random fragments withing them.

Reply:
@peppelinux not sure how you read your proposal. The issue you are referring to suggests randomized request URIs, that's inline with what @jogu suggests, right?
@jogu do you (implcity) answer my question whether we should assume the transaction state is carried through the request URI?

Reply:
I think the transaction state is carried over by the request URI. Another option would be that the request_uri POST endpoint would also expect a `state` parameter if `state` was present in the authz request.

Reply:
> An adversary that has obtained/steal the RP's private key would be able to issue its own signed requests at its own request_uri endpoint, as well as hijack presentations to some evil response_uri endpoint.

But isn't a undetected key compromise of the RP a much bigger problem? The key is there to trust the RP in case the `client_id_scheme` requires request signing.

If possible, I would not put the `request_uri` in the `client_metadata` since we would lose the JAR compatibility. It would only help if the `client_metadata` is not provided in the authz request.

Reply:
I believe @peppelinux is suggesting adding metadata should that request_uri from the verifier has to match some kind of pattern. This would mean it can't be abused to get the wallet to fetch arbitrary urls. It might be a good change to make but I would say (as it affects both the get & post mode) it should be discussed in a separate issue.

Reply:
sounds like OAuth redirect URI pattern matching, something that has turned out to be brittle, we git rid of it with the Security BCP

Reply:
WG call: will adopt suggested change and open a new issue to discuss (potential) `state` parameter